src/commissaire_http/authentication/__init__.py

```diff
@@ -67,3 +68,30 @@ def authenticate(self, environ, start_response):
         :rtype: bool
         """
         return False
+
+
+def decode_basic_auth(logger, http_auth):
+    """
+    Decodes basic auth from the header string.
+
+    :param logger: logging instance
+    :type logger: Logger
+    :param http_auth: Basic authentication header
+    :type http_auth: string
+    :returns: tuple -- (username, passphrase) or (None, None) if empty.
+    :rtype: tuple
+    """
+    if http_auth is not None:
+        if http_auth.lower().startswith('basic '):
+            try:
+                decoded = tuple(base64.decodebytes(
+                    http_auth[6:].encode('utf-8')).decode().split(':'))
+                if logger:
+                    logger.debug('Credentials given: {0}'.format(decoded))
```

Comment:
Almost certainly from our original code but just a reminder that we no longer need to use ``{N}`` with Python 3. It won't hurt anything to use it and it's not worth changing but, when new code is written using ``{}`` is considered preferable.